github/status_test.go

```diff
@@ -15,22 +16,35 @@ import (
 
 func TestGitHubStatusUseMockAPI(t *testing.T) {
 	cfg := help.FakeTestCfg
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusCreated)
-		fmt.Fprintln(w, "Anything goes...")
-	}))
-	defer ts.Close()
-
 	context := "cogito/test"
 	targetURL := "https://cogito.invalid/builds/job/42"
 	desc := time.Now().Format("15:04:05")
 	state := "success"
+	var testCases = []struct {
+		name       string
+		wantErr    error
+		wantStatus int
+		write      string
+	}{
+		{"Success: no errors",
+			nil, http.StatusCreated, "Anithing goes..."},
+		{"Api Fails: Internal Server Error",
+			fmt.Errorf("Internal Server Error"), http.StatusInternalServerError, "Something bad happened!"},
+	}
+	for _, tc := range testCases {
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(tc.wantStatus)
+			fmt.Fprintln(w, tc.write)
+		}))
+		defer ts.Close()
 
-	status := github.NewStatus(ts.URL, cfg.Token, cfg.Owner, cfg.Repo, context)
-	err := status.Add(cfg.SHA, state, targetURL, desc)
-
-	if err != nil {
-		t.Fatalf("\ngot:  %v\nwant: no error", err)
+		status := github.NewStatus(ts.URL, cfg.Token, cfg.Owner, cfg.Repo, context)
+		err := status.Add(cfg.SHA, state, targetURL, desc)
+		if err != nil {
+			if !strings.Contains(err.Error(), fmt.Sprintf("%s\nMay be %s is not healthy?", tc.wantErr.Error(), ts.URL)) {
```

Comment:
I don't like much the idea of having this if statement specific to one of the test case in a loop over all test cases.

Either you write it in a way that is generic enough by pushing what's needed to `testCases` and the loop would make sense, i.e `testCases` can be extended later to cover more status code and error message. Or you just write a test function for each case and forget about the loop.

Reply:
At the beginning I started writing another test, with this implementation I try to avoid duplication.

Reply:
> the loop would make sense, i.e testCases can be extended later to cover more status code and error message

This this also my motivation, let's do more generic.